File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/Vega/Services/Vehicles/Concrete/VehiclesStatisticsService.cs
namespace QuartzEnergy.FracSchedule.Services.Vega.Services.Vehicles.Concrete
{
    using System.Threading.Tasks;

    using QuartzEnergy.Common.Dal.Infrastructure;
    using QuartzEnergy.Common.Services.Models.Statistics;
    using QuartzEnergy.Common.Services.Services.Statistics;
    using QuartzEnergy.FracSchedule.Domain.Entities;
    using QuartzEnergy.FracSchedule.Services.Vega.Mappers;
    using QuartzEnergy.FracSchedule.Services.Vega.Models.Vehicles.Statistics;

    public sealed class VehiclesStatisticsService
        : NoRequestStatisticsService<VehiclesByMakerData>, IVehiclesStatisticsService
    {
        public VehiclesStatisticsService(IUnitOfWorkFactory unitOfWorkFactory)
            : base(unitOfWorkFactory)
        {
        }

        protected override async Task<Statistics<EmptyStatisticsRequest, VehiclesByMakerData>> DoGetStatistics(IUnitOfWork uow, EmptyStatisticsRequest request)
        {
            var vehiclesRep = uow.GetRepository<VehicleEntity>();
            var makersRep = uow.GetRepository<MakerEntity>();
            var modelsRep = uow.GetRepository<ModelEntity>();

            return await VehiclesMapper.MapToStatisticsByMakers(
                       vehiclesRep.GetAll(),
                       makersRep.GetAll(),
                       modelsRep.GetAll());
        }
    }
}

File: quartz-energy/src/frontend/src/app/vega/services/vehicles/statistics/concrete/vehicles-statistics.service.ts
import { IVehiclesStatisticsService } from './../i-vehicles-statistics-service';
import { VehiclesByMakerData } from './../../../../models/vehicles/statistics/vehicles-by-maker-data';
import { Injectable, Inject } from '@angular/core';
import { NoRequestStatisticsService } from '../../../../../common/services/statistics/concrete/no-request-statistics.service';
import { IConfigServiceToken, IConfigService } from '../../../../../common/services/config/i-config-service';

@Injectable()
export class VehiclesStatisticsService
  extends NoRequestStatisticsService<VehiclesByMakerData>
  implements IVehiclesStatisticsService {
    constructor (
      @Inject(IConfigServiceToken) private configService: IConfigService) {
      super();
    }

    protected getBaseUrl(): string {
        return this.configService.getApiBaseUrl() + 'vehicles/statistics/bymakers';
    }
}
